File: scaffold/src/main/java/top/xuqingquan/web/system/MiddlewareWebClientBase.java
package top.xuqingquan.web.system;

import android.webkit.WebViewClient;

public class MiddlewareWebClientBase extends WebViewClientDelegate {
    private MiddlewareWebClientBase mMiddleWrareWebClientBase;

    @SuppressWarnings("unused")
    public MiddlewareWebClientBase(MiddlewareWebClientBase client) {
        super(client);
        this.mMiddleWrareWebClientBase = client;
    }

    @SuppressWarnings("WeakerAccess")
    protected MiddlewareWebClientBase(WebViewClient client) {
        super(client);
    }

    @SuppressWarnings("WeakerAccess")
    protected MiddlewareWebClientBase() {
        super(null);
    }

    public final MiddlewareWebClientBase next() {
        return this.mMiddleWrareWebClientBase;
    }

    @Override
    public final void setDelegate(WebViewClient delegate) {
        super.setDelegate(delegate);

    }

    @SuppressWarnings("UnusedReturnValue")
    public final MiddlewareWebClientBase enq(MiddlewareWebClientBase middleWrareWebClientBase) {
        setDelegate(middleWrareWebClientBase);
        this.mMiddleWrareWebClientBase = middleWrareWebClientBase;
        return middleWrareWebClientBase;
    }

}

File: scaffold_web/src/main/java/top/xuqingquan/web/system/MiddlewareWebClientBase.kt
package top.xuqingquan.web.system

import android.webkit.WebViewClient

open class MiddlewareWebClientBase : WebViewClientDelegate {
    private var mMiddleWareWebClientBase: MiddlewareWebClientBase? = null

    override var delegate: WebViewClient?
        get() = super.delegate
        set(delegate) {
            super.delegate = delegate

        }

    constructor(client: MiddlewareWebClientBase) : super(client) {
        this.mMiddleWareWebClientBase = client
    }

    protected constructor(client: WebViewClient?) : super(client)

    protected constructor() : super(null)

    operator fun next(): MiddlewareWebClientBase? {
        return this.mMiddleWareWebClientBase
    }

    fun enq(middleWrareWebClientBase: MiddlewareWebClientBase): MiddlewareWebClientBase {
        super.delegate = middleWrareWebClientBase
        this.mMiddleWareWebClientBase = middleWrareWebClientBase
        return middleWrareWebClientBase
    }

}
